src/core/services/theming/theming.js

```diff
@@ -538,6 +539,36 @@ function ThemingDirective($mdTheming, $interpolate, $log) {
   };
 }
 
+/**
+ * Special directive that will disable ALL runtime Theme style generation and DOM injection
+ *
+ * <link rel="stylesheet" href="angular-material.min.css">
+ * <link rel="stylesheet" href="angular-material.themes.css">
+ *
+ * <body md-theme-css>
+ *  ...
+ * </body>
+ *
+ * Note: Using md-themes-css directive requires the developer to load external
+ * theme stylesheets; e.g. custom themes from Material-Tools:
+ *
+ *       `angular-material.themes.css`
+ *
+ * Another option is to use the ThemingProvider to configure and disable the attribute
+ * conversions; this would obviate the use of the `md-themes-css` directive
+ *
+ */
+function disableThemesDirective() {
+  return {
```

Comment:
I think that you can avoid the `compile` by doing something like this.:

```js
function disableThemesDirective() {
  disableTheming = true;

  return {
    restrict : 'A',
    priority : '900'
  };
}
```

The directive function gets called only once, the first time it matches an element.